### ISSUE TITLE
Windows Hello login issue

### ISSUE THREAD

**Issue by wipeer** [gazugafan/fingerpass#1]
As the FingerPass is running in the background, when my computer gets locked I am unable to use the fingerprint scanner to unlock the Windows. The fingerprint scanner is used by FingerPass so Windows Hello does not work. In fact, FingerPass saying in "bubble" that the program was not recognized.

**Comment by gazugafan** [comment]
I did program in a mechanism that should make FingerPass release focus of your scanner when you logout. I have seen this problem occasionally myself, though. I'm guessing I need to handle more events that lead to Windows locking besides just logging out. I'll look into it!

Not sure what you mean by FingerPass saying in "bubble" that the program was not recognized. Do you have a screenshot?

**Comment by wipeer** [comment]
Thank you!

By bubble I mean notification bubble by Windows:
[https://i.ibb.co/WH3g6cw/bubble.gif](url)

**Comment by wipeer** [comment]
Thank you!

By bubble, I mean notification bubble by Windows:
![bubble](https://user-images.githubusercontent.com/11052199/140887067-15b516b7-c357-4f78-a030-d2699e220bd2.gif)

**Comment by gazugafan** [comment]
Oh, interesting! I'll look into this soon.

**Comment by michelpb941004** [comment]
Actually i have the same problem since a while, how can i fix it?,, do u have any news about it?

**Comment by gazugafan** [comment]
Thanks for the report! It's just a really low-priority bug for me to devote time to. I'll try to get around to it soon, though :)

In the meantime, the workaround would be to right-click the fingerpass tray icon and click Pause Fingerprinting before logging out of Windows. That forces fingerpass to give up control of the fingerprint scanner, so Windows Hello etc can use it again.

**Comment by AtmanActive** [comment]
This issue is now becoming a big problem these days as more and more apps add support for windows hello master password unlocking themselves. As long as Fingerpass uses the reader exclusively, it is unusuable as it blocks all other programs from working correctly. Bummer, it was such a great idea and a decent implementation to boot.

**Comment by AtmanActive** [comment]
There could be a solution though: just release the fingerprint scanner programmatically until you detect a window with a defined title for passwords we have in our database. When window is detected, then, take control of the fingerprint scanner and listen for touch events. When window goes out of focus, release the fingerprint scanner again. That's assuming that fingerprint scanner access can be acquired/released quickly and painlessly.
Thanks.

**Comment by gazugafan** [comment]
> There could be a solution though: just release the fingerprint scanner programmatically until you detect a window with a defined title for passwords we have in our database. When window is detected, then, take control of the fingerprint scanner and listen for touch events. When window goes out of focus, release the fingerprint scanner again. That's assuming that fingerprint scanner access can be acquired/released quickly and painlessly. Thanks.

This is a **_very good_** idea. As long as the constant checking of the window title and releasing/focusing the scanner isn't a performance problem, I don't see why this wouldn't work. 

It's really hard for me to devote any spare time to this project at the moment, though. If you're able to implement this, PR's welcome! Otherwise, if/when I get back to this project this will be what I tackle first.

**Comment by Giulioski** [comment]
I don't think this is a good approach. You should take and release the scanner so long filter "app AND windows title = True" Else release scanner this is by far much more easier than adding an exclusion for every app or Microsoft change on the Hello interface. Could you add a parameter read-in to Fingerpass? For example /PON and /POFF (to set Pause on or off)? this could be usefull for others extendings possibilities

**Comment by gazugafan** [comment]
> You should take and release the scanner so long filter "app AND windows title = True" Else release scanner

I think this is what was proposed. Or if not, this is how I read it and would implement it.

> Could you add a parameter read-in to Fingerpass? For example /PON and /POFF (to set Pause on or off)? this could be usefull for others extendings possibilities

I don't see why not. Could you create another issue for this? That way if I ever get back to work on Fingerpass, I'll have it logged and know what to work on.

**Comment by Giulioski** [comment]
> This is a **_very good_** idea. As long as the constant checking of the window title and releasing/focusing the scanner isn't a performance problem, I don't see why this wouldn't work.

Years ago I wrote a little test app for polling window titles and it had no footprint on cpu performance
